backend/common/compute/estimate_distribution.py

```diff
@@ -0,0 +1,61 @@
+import numba
+import concurrent.futures
+import numpy as np
+from scipy import sparse
+from backend.common.constants import XApproxDistribution
+
+
+@numba.njit(fastmath=True, error_model="numpy", nogil=True)
+def min_max(arr):
+    """Return (min, max) values for the ndarray."""
+    n = arr.size
+    odd = n % 2
+    if not odd:
+        n -= 1
+    max_val = min_val = arr[0]
+    i = 1
+    while i < n:
+        x = arr[i]
+        y = arr[i + 1]
+        if x > y:
+            x, y = y, x
+        min_val = min(x, min_val)
+        max_val = max(y, max_val)
+        i += 2
+    if not odd:
+        x = arr[n]
+        min_val = min(x, min_val)
+        max_val = max(x, max_val)
+    return min_val, max_val
+
+
+def estimate_approximate_distribution(X) -> XApproxDistribution:
+    """
+    Estimate the distribution (normal, exponential) of the X matrix.
+
+    Currently this is based upon the assumption that scRNA-seq data is
+    exponentially distributed in its raw (count) form, and when logged,
+    any (max-min) range in excess of 24 is extremely unlikely.
```

Comment:
```suggestion
    any (max-min) range in excess of 24 implies tens of millions of 
    observations of a single feature and so is extremely unlikely.
```
Suggest a note to explain explicitly why it is unlikely.